### PR TITLE
Add introduction to authorization section.

### PR DESCRIPTION
**Pull request by msporny**
This PR is a partial alternate to PR #226. This particular PR is a non-normative introductory section to the Authorization section in the specification. It sets expectations, but delegates the normative language to subsections (in separate PRs). It differs from PR #226 in a few ways:

* It is only the introduction, the specific normative sections on authz mechanisms will be in separate PRs.
* It does not say anything about RAR (separate PR).
* It does not say anything about OAuth2.0 (separate PR).
* It does not say anything about delegation (separate PR).
* It does not say anything about GNAP (separate PR).
* It does not say anything about forbidden methods (separate PR).

***
<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-http-api/pull/228.html" title="Last updated on Sep 25, 2021, 11:23 PM UTC (006a615)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-http-api/228/41dbd3e...006a615.html" title="Last updated on Sep 25, 2021, 11:23 PM UTC (006a615)">Diff</a>

### CODE REVIEW

**Review by OR13 — APPROVED** [review]

**Review by mprorock — APPROVED** [review]

**Review by mkhraisha — APPROVED** [review]

**Review by peacekeeper — APPROVED** [review]
I have been mostly quiet on the recent authorization discussions, but since I'm listed as one of the code owners in this repo, and I'm also an implementer, I'll write a few lines here.

I sympathize quite a bit with Adrian's comments in the last few weeks/months. What I have heard in the various calls and threads is that "the vendors want to do X", but "what's good for the individual is Y". I admit I don't know the details of the different proposals well enough to understand the implications (e.g. OAuth2 = digital slavery? GNAP/RAR = sovereignty?), but I do think Adrian's point is valid that we shouldn't just necessarily do what's easiest for vendors. Reminds me of the did:web discussion, which is great for vendors but doesn't really offer individidual sovereignty.

As a vendor implementing this VC HTTP API, we (Danube Tech) haven't implemented any authorization mechanism. So far we have considered this to not be part of the API but to happen elsewhere, e.g. in external API gateways, proxies, firewall rules, authenticated TLS connections, etc. I think that other implementers have thought about this more thoroughly than we have, therefore I haven't been too involved in the discussions about specific technology choices.

As a code owner, I think Manu has done a nice job with this and the subsequent PRs to break up individual topics into more manageable pieces, so I suggest we use this as a basis.

File: index.html [hunk 1]
```diff
@@ -272,11 +272,11 @@ <h2>
       </p>
 
      <p>
-A <dfn>conforming VC HTTP API client</dfn> is ...
+A conforming <dfn>VC HTTP API client</dfn> is ...
```
Comment (jandrieu) [inline index.html:275]:
```suggestion
A conforming <dfn>VC API client</dfn> is ...
```

File: index.html [hunk 2]
```diff
@@ -272,11 +272,11 @@ <h2>
...
      </p>
 
       <p>
-A <dfn>conforming VC HTTP API server</dfn> is ...
+A conforming <dfn>VC HTTP API server</dfn> is ...
```
Comment (jandrieu) [inline index.html:279]:
```suggestion
A conforming <dfn>VC API server</dfn> is ...
```

File: index.html [hunk 3]
```diff
@@ -296,7 +296,20 @@ <h2>The VC HTTP API</h2>
     <section>
       <h3>Authorization</h3>
       <p>
+The VC HTTP API can be deployed in a variety of networking environments which
```
Comment (jandrieu) [inline index.html:299]:
```suggestion
The VC API can be deployed in a variety of networking environments which
```

File: index.html [hunk 4]
```diff
@@ -296,7 +296,20 @@ <h2>The VC HTTP API</h2>
...
+might contain hostile actors. As a result, conforming <a>VC HTTP API servers</a>
+require conforming <a>VC HTTP API clients</a> to utilize secure authorization
```
Comment (jandrieu) [inline index.html:301]:
```suggestion
require conforming <a>VC API clients</a> to utilize secure authorization
```

**Review by OR13 — CHANGES_REQUESTED** [review]
I require @jandrieu 's proposed changes to be accepted to approve.

**Review by jandrieu — CHANGES_REQUESTED** [review]
Just the name thing.

**Comment by msporny** [comment]
All change requests merged, no objections, merging.

**Pull request merged**